lib/ansible/modules/cloud/cloudstack/cs_vlan_ip_range.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2018, David Passante <@dpassante>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cs_vlan_ip_range
+short_description: Manages VLAN IP ranges on Apache CloudStack based clouds.
+description:
+    - Create and delete VLAN IP range.
+version_added: '2.8'
+author: "David Passante (@dpassante)"
+options:
+  network:
+    description:
+      - The network name or id.
+    required: true
+  start_ip:
+    description:
+      - The beginning IPv4 address in the VLAN IP range.
+      - Only considered on create.
+    required: true
+  end_ip:
+    description:
+      - The ending IPv4 address in the VLAN IP range.
+      - If not specified, value of C(start_ip) is used.
+      - Only considered on create.
+  gateway:
+    description:
+      - The gateway of the VLAN IP range.
+      - Only considered on create.
+    required: true
```

Comment:
seems gateway and netmask are only required in case of state=present, are they? In this case i would suggest to:

~~~diff
     argument_spec.update(dict(
         network=dict(type='str', required=True),
         zone=dict(type='str'),
         start_ip=dict(type='str', required=True),
         end_ip=dict(type='str'),
-        gateway=dict(type='str', required=True),
-        netmask=dict(type='str', required=True),
+        gateway=dict(type='str'),
+        netmask=dict(type='str'),
         start_ipv6=dict(type='str'),
         end_ipv6=dict(type='str'),
         gateway_ipv6=dict(type='str'),
         vlan=dict(type='str'),
         state=dict(choices=['present', 'absent'], default='present'),
         domain=dict(type='str'),
         account=dict(type='str'),
         project=dict(type='str'),
         for_virtual_network=dict(type='bool', default=False),
         for_systemvms=dict(type='bool', default=False),
     ))


     module = AnsibleModule(
         argument_spec=argument_spec,
         required_together=cs_required_together(),
         mutually_exclusive=(
             ['account', 'project'],
         ),
+        required_if=(("state", "present", ("gateway", "netmask")),),
         supports_check_mode=True,
    )
~~~

Reply:
Right! Only the `start_ip` and `end_ip` parameters are used in other cases.